packages/server/modules/workspaces/helpers/roles.ts

```diff
@@ -0,0 +1,56 @@
+import { Roles, StreamRoles, WorkspaceRoles } from '@speckle/shared'
+import { WorkspaceAcl } from '@/modules/workspacesCore/domain/types'
+import { WorkspaceRole } from '@/modules/core/graph/generated/graphql'
+
+export const isUserLastWorkspaceAdmin = (
+  workspaceRoles: WorkspaceAcl[],
+  userId: string
+): boolean => {
+  const workspaceAdmins = workspaceRoles.filter(
+    ({ role }) => role === Roles.Workspace.Admin
+  )
+  const isUserAdmin = workspaceAdmins.some((role) => role.userId === userId)
+
+  return isUserAdmin && workspaceAdmins.length === 1
+}
+
+/**
+ * Given a user's workspace role, return the role they should have for workspace projects.
+ */
+export const mapWorkspaceRoleToProjectRole = (
+  workspaceRole: WorkspaceRoles
+): StreamRoles => {
+  switch (workspaceRole) {
+    case Roles.Workspace.Guest:
+    case Roles.Workspace.Member:
+      return Roles.Stream.Reviewer
+    case Roles.Workspace.Admin:
+      return Roles.Stream.Owner
+  }
+}
+
+export const mapProjectRoleToWorkspaceRole = (
```

Comment:
Does this make sense? When a user gets invited to a workspace project, and has to be also added to the workspace once the invite is accepted - what kind of project role should they get?

Reply:
discussed a bit in standup the other day - I plan to spend a little time tomorrow writing down more specific requirements for project/workspace roles